Sources/SwiftDocCUtilities/Action/Actions/MergeAction.swift

```diff
@@ -0,0 +1,164 @@
+        
+        let targetURL = try Self.createUniqueDirectory(inside: fileManager.temporaryDirectory, template: firstArchive, fileManager: fileManager)
+        defer {
+            try? fileManager.removeItem(at: targetURL)
+        }
+      
+        // TODO: Merge the LMDB navigator index
+        
+        let jsonIndexURL = targetURL.appendingPathComponent("index/index.json")
+        guard let jsonIndexData = fileManager.contents(atPath: jsonIndexURL.path) else {
+            throw CocoaError.error(.fileReadNoSuchFile, userInfo: [NSFilePathErrorKey: jsonIndexURL.path])
+        }
+        var combinedJSONIndex = try JSONDecoder().decode(RenderIndex.self, from: jsonIndexData)
+        
+        for archive in archives.dropFirst() {
+            for directoryToCopy in ["data/documentation", "data/tutorials", "documentation", "tutorials", "images", "videos", "downloads"] {
+                let fromDirectory = archive.appendingPathComponent(directoryToCopy, isDirectory: true)
+                let toDirectory = targetURL.appendingPathComponent(directoryToCopy, isDirectory: true)
+                
+                for from in (try? fileManager.contentsOfDirectory(at: fromDirectory, includingPropertiesForKeys: nil, options: .skipsHiddenFiles)) ?? [] {
+                    try fileManager.copyItem(at: from, to: toDirectory.appendingPathComponent(from.lastPathComponent))
+                }
+            }
+            guard let jsonIndexData = fileManager.contents(atPath: archive.appendingPathComponent("index/index.json").path) else {
+                throw CocoaError.error(.fileReadNoSuchFile, userInfo: [NSFilePathErrorKey: archive.appendingPathComponent("index/index.json").path])
+            }
+            let renderIndex = try JSONDecoder().decode(RenderIndex.self, from: jsonIndexData)
+            
+            try combinedJSONIndex.merge(renderIndex)
+        }
+        
+        try fileManager.createFile(at: jsonIndexURL, contents: RenderJSONEncoder.makeEncoder(emitVariantOverrides: false).encode(combinedJSONIndex))
+        
+        // TODO: Build landing page from input or synthesize default landing page
+        
+        // TODO: Inactivate external links outside the merged archives
+        
+        try Self.moveOutput(from: targetURL, to: outputURL, fileManager: fileManager)
+        
+        return ActionResult(didEncounterError: false, outputs: [outputURL])
+    }
+    
+    private func validateThatArchivesHaveDisjointData() throws {
+        // Check that the archives don't have overlapping data
+        typealias ArchivesByDirectoryName = [String: Set<String>]
+        
+        var archivesByTopLevelDocumentationDirectory = ArchivesByDirectoryName()
+        var archivesByTopLevelTutorialDirectory = ArchivesByDirectoryName()
+        
+        // Gather all the top level /data/documentation and /data/tutorials directories to ensure that the different archives don't have overlapping data
+        for archive in archives {
```

Comment:
Yes, I did it this way because I only wanted to check the if more than one archive had the same top-level module or technology. It's not intended to be supported to merge multiple archives that contain the same modules and/or technologies so checking the top-level directories is sufficient. 

I updated the to code do two nested loops to gather all the unique module names and technology names inside the data directory's direct subdirectories.